test/unit/os/env_spec.lua

```diff
@@ -305,4 +311,12 @@ describe('env.c', function()
       eq(output[4], 0)
     end)
   end)
+
+  describe('init_homedir', function()
+    itp('able to get an non-empty homedir with unset $HOME environment variable', function()
+      os_unsetenv('HOME')
```

Comment:
looks like you have good coverage in the functional tests (not unit/FFI test). And this unit test doesn't seem very important. So let's remove this one, then we can remove the `#ifdef UNIT_TESTING` code. That eliminates a lot of changes.

Reply:
Done